project_leverage/controllers/main.py

```diff
@@ -0,0 +1,101 @@
+# Copyright (C) 2022 Open Source Integrators
+# Copyright (C) 2020 Serpent Consulting Services
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo.http import request
+
+from odoo.addons.sale_timesheet.controllers.portal import SaleTimesheetCustomerPortal
+
+DEFAULT_MONTH_RANGE = 3
+
+
+class SaleTimesheetController(SaleTimesheetCustomerPortal):
+    def _plan_prepare_values(self, projects):
+        employee_obj = request.env["hr.employee"]
+        values = super(SaleTimesheetController, self)._plan_prepare_values(projects)
+        _select = self.select_line()
+        _from = self.line_from()
+        _where, param = self.line_where(projects)
+        _group = self.line_group()
+        query = _select + _from + _where + _group
+        request.env.cr.execute(query, param)
+        raw_data = request.env.cr.dictfetchall()
+        project_data = {}
+        month_list = []
+        month_name_list = []
+        for rec in raw_data:
+            month_name_list.append(rec.get("month_year"))
+            if (
+                str(rec.get("date_mm")).strip() + str(int(rec.get("year")))
+                not in month_list
+            ):
+                month_list.append(
+                    str(rec.get("date_mm")).strip() + str(int(rec.get("year")))
+                )
+            if project_data.get(rec.get("employee_id") or 0):
+                project_data[rec.get("employee_id") or 0].update(
+                    {
+                        str(rec.get("date_mm")).strip()
+                        + str(int(rec.get("year"))): {
```

Comment:
again this concatenation -> please use an intermediate variable with a meaningful name.